optimum/onnxruntime/configuration.py

```diff
@@ -730,14 +732,87 @@ def create_fusion_options(self, model_type: str) -> FusionOptions:
         for attr_name, fusion_attr_name in attribute_map.items():
             setattr(args, fusion_attr_name, getattr(self, attr_name))
 
-        for attr, value in self.__dict__:
+        for attr, value in self.__dict__.items():
             if hasattr(args, attr):
                 continue
             setattr(args, attr, value)
 
         return FusionOptions.parse(args)
 
 
+class AutoOptimizationConfig:
+    _LEVELS = [
+        {
```

Comment:
I feel like 6 levels is a lot. Here is a proposal:
- 0: `general_optimization_level=1` and that's it (basic),
- 1: `general_optimization_level=2`, `enable_transformers_specific_optimizations=True` and `enable_gelu_approximation=True` (medium),
- 2: `general_optimization_level=99`, `enable_transformers_specific_optimizations=True`, `enable_gelu_approximation=True` and `fp16=True` (advanced).

I would also maybe name them O0, O1 and O2 to avoid any confusion with ORT optimization levels. WDYT?

Reply:
I agree!
What is the expected drop in accuracy with `enable_gelu_approximation=True`? If it's significant, I would suggest adding one level with only `general_optimization_level=2, enable_transformers_specific_optimizations=True `

Reply:
An `AutoOptimizationConfig` could be interesting in the case where we want to optimize a model while keeping it compatible with quantization or if already have a quantized model and want to automatically define the optimizations that can be applied on it. I think we should wait before adding it as the different options could vary as there is no clear options which currently makes more sense than the others to me

Reply:
The current `OptimizationConfig` is not intuitive, hence the `AutoOptimizationConfig` hiding those details to the user. 
Also, we can add those features later, for now the `AutoOptimizationConfig` provides "good" common optimizations, independent from quantization, exactly as what `AutoQuantizationConfig` does for quantization.